### PR TITLE
IntegrationEvent adds retry

### PR DESCRIPTION
**Pull request by zhenlei520**

### CODE REVIEW

File: src/Dispatcher/MASA.BuildingBlocks.Dispatcher.IntegrationEvents/Logs/IIntegrationEventLogService.cs [hunk 1]
```diff
@@ -1,7 +1,14 @@
...
 public interface IIntegrationEventLogService
 {
-    Task<IEnumerable<IntegrationEventLog>> RetrieveEventLogsPendingToPublishAsync(Guid transactionId);
+    /// <summary>
+    /// Get messages to retry
+    /// </summary>
+    /// <param name="retryBatchSize">The size of a single event to be retried</param>
+    /// <param name="maxRetryTimes"></param>
+    /// <returns></returns>
+    Task<IEnumerable<IntegrationEventLog>> RetrieveEventLogsFailedToPublishAsync(int retryBatchSize, int maxRetryTimes);
```
Comment (doddgu) [inline src/Dispatcher/MASA.BuildingBlocks.Dispatcher.IntegrationEvents/Logs/IIntegrationEventLogService.cs:11]: why not set a default value.

File: src/Dispatcher/MASA.BuildingBlocks.Dispatcher.IntegrationEvents/Logs/IIntegrationEventLogService.cs [hunk 2]
```diff
@@ -10,4 +17,13 @@ public interface IIntegrationEventLogService
...
     Task MarkEventAsFailedAsync(Guid eventId);
+
+    /// <summary>
+    /// Delete successfully published and expired data
+    /// </summary>
+    /// <param name="expireDate"></param>
+    /// <param name="batchCount"></param>
+    /// <param name="token"></param>
+    /// <returns></returns>
+    Task DeleteExpiresAsync(DateTime expireDate, int batchCount = 1000, CancellationToken token = default);
```
Comment (doddgu) [inline src/Dispatcher/MASA.BuildingBlocks.Dispatcher.IntegrationEvents/Logs/IIntegrationEventLogService.cs:28]: expireDate or expireTime or expires?

File: src/Dispatcher/MASA.BuildingBlocks.Dispatcher.IntegrationEvents/Logs/IntegrationEventLog.cs [hunk 3]
```diff
@@ -33,13 +36,10 @@ public IntegrationEventLog(IIntegrationEvent @event, Guid transactionId) : this(
...
-        State = IntegrationEventStates.NotPublished;
-        TimesSent = 0;
+        Content = System.Text.Json.JsonSerializer.Serialize((object) @event);
         TransactionId = transactionId;
     }
 
-
     public IntegrationEventLog DeserializeJsonContent(Type type)
     {
         Event = (System.Text.Json.JsonSerializer.Deserialize(Content, type) as IIntegrationEvent)!;
```
Comment (doddgu) [inline src/Dispatcher/MASA.BuildingBlocks.Dispatcher.IntegrationEvents/Logs/IntegrationEventLog.cs:45]: fix `Unchanged files with check annotations`

File: src/Dispatcher/MASA.BuildingBlocks.Dispatcher.IntegrationEvents/Logs/IIntegrationEventLogService.cs (hunk 2 above)
Comment (zhenlei520) [reply src/Dispatcher/MASA.BuildingBlocks.Dispatcher.IntegrationEvents/Logs/IIntegrationEventLogService.cs:28]: How about ExpiresAt？

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**